File: backend/app/schemas/analytics.py
from datetime import date
from decimal import Decimal
from pydantic import BaseModel
import uuid


class DashboardSummary(BaseModel):
    total_students: int
    total_courses: int
    avg_grade: Decimal
    active_users_today: int
    at_risk_count: int


class EngagementPoint(BaseModel):
    date: date
    logins: int
    lesson_views: int
    submissions: int


class EngagementReport(BaseModel):
    points: list[EngagementPoint]


class AtRiskStudent(BaseModel):
    student_id: uuid.UUID
    student_name: str
    email: str
    course_average: Decimal
    last_login: date | None
    missing_assignments: int


class GradeDistributionBucket(BaseModel):
    label: str  # A, B, C, D, F
    count: int
    percentage: Decimal


class GradeDistribution(BaseModel):
    course_id: uuid.UUID
    buckets: list[GradeDistributionBucket]
    mean: Decimal
    median: Decimal

File: frontend/src/types/analytics.ts
export interface DashboardSummary {
  total_students: number
  total_courses: number
  avg_grade: number
  active_users_today: number
  at_risk_count: number
}

export interface EngagementPoint {
  date: string
  logins: number
  lesson_views: number
  submissions: number
}

export interface EngagementReport {
  points: EngagementPoint[]
}

export interface GradeDistributionBucket {
  label: string
  count: number
  percentage: number
}

export interface GradeDistribution {
  course_id: string
  buckets: GradeDistributionBucket[]
  mean: number
  median: number
}
